### PR TITLE
Make MaterialTextField's BackgroundColor works with a global style

### PR DESCRIPTION
**Pull request by softlion**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Bug fix

### :arrow_heading_down: What is the current behavior?

When setting the BackgroundColor property of a MaterialTextField in a Style in App.xaml, it is ignored.

### :new: What is the new behavior (if this is a feature change)?

It is applied as expected.
This is not a feature change.

### :boom: Does this PR introduce a breaking change?

No

### :bug: Recommendations for testing

Sample project updated: a global style has been added and the sample page for MaterialTextField has been udpated.

I renamed the override of BackgroundColor to RealBackgroundColor, but i also detected all changes to BackgroundColor and they are applied to the RealBackgroundColor property before forcing the BackgroundColor to transparent.

The issue was caused by this override of BackgroundColor. When the style is set in App.xaml, BackgroundColor was applied to the base BackgroundColor property, not the override. This is because the override never overrides when the control is casted to a base class. And this is why using "new" to override is a very bad practice.

### :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines 
- [ ] Relevant documentation was updated
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by martijn00** [comment]
@softlion sorry, can you rebase this too?

**Comment by softlion** [comment]
done

**Comment by softlion** [comment]
done. I was unable to rebase with visual studio. some kind of bug in this git ui. so i merged.

File: XF.Material/UI/MaterialTextField.xaml.cs [hunk 1]
```diff
@@ -193,10 +193,10 @@ public bool AlwaysShowUnderline
         /// <summary>
         /// Gets or sets the background color of this text field.
         /// </summary>
-        public new Color BackgroundColor
+        public Color RealBackgroundColor
```
Comment (martijn00) [inline XF.Material/UI/MaterialTextField.xaml.cs]: Not sure if i like the name RealBackgroundColor... Do you have better ideas instead of `Real`?

**Review by martijn00 — CHANGES_REQUESTED** [review]

**Comment by softlion** [comment]
They have the same issue on a xamarin forms control out of the box. 
They called it Color.
Not sure it is better ...
It could be CardBackgroudColor.

**Comment by martijn00** [comment]
I like `CardBackgroudColor`. Lets go for that.

**Comment by softlion** [comment]
done

**Pull request merged**